### PR TITLE
[DEV-1187] improvements

### PR DESCRIPTION
**Pull request by Krysh90**
missing:
- mobile check
- styled input types

### CODE REVIEW

File: src/stories/form/StyledModalDropdown.tsx [hunk 1]
```diff
@@ -53,33 +56,49 @@ export default function StyledModalDropdown<T>({
...
+            <button
+              className="flex justify-between border border-dfxGray-400 text-base font-normal rounded-md px-4 py-2 shadow-sm w-full"
+              onClick={() => setShowModal(true)}
+              onBlur={onBlur}
+              {...props}
+            >
+              <div className="flex flex-col justify-between text-left gap-1">
+                {value ? (
+                  <>
+                    {descriptionFunc && descriptionFunc(value) && (
```
Comment (davidleomay) [inline src/stories/form/StyledModalDropdown.tsx:82]: `descriptionFunc?.(value)`

File: src/stories/form/StyledModalDropdown.tsx [hunk 2]
```diff
@@ -53,33 +56,49 @@ export default function StyledModalDropdown<T>({
...
+                      <span className="text-dfxGray-800 text-xs h-min leading-none">{descriptionFunc(value)}</span>
+                    )}
+                    <span
+                      className={'text-dfxBlue-800 leading-none font-base'.concat(
+                        descriptionFunc && descriptionFunc(value) ? '' : ' py-2',
```
Comment (davidleomay) [inline src/stories/form/StyledModalDropdown.tsx:87]: `descriptionFunc?.(value)`

File: src/components/tabs/buy-tab-content/buy.process.tsx [hunk 3]
```diff
@@ -52,14 +50,24 @@ export function BuyTabContentProcess({ asset, onBack }: BuyTabContentProcessProp
   const {
     control,
     handleSubmit,
+    setValue,
+    resetField,
     formState: { errors },
   } = useForm<FormData>({ defaultValues: { asset } });
   const data = useWatch({ control });
   const validatedData = validateData(useDebounce(data, 500));
+  const selectedBankAccount = useWatch({ control, name: 'bankAccount' });
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply src/components/tabs/buy-tab-content/buy.process.tsx:59]: only with `as BankAccount`

File: src/components/tabs/buy-tab-content/buy.process.tsx [hunk 4]
```diff
@@ -71,10 +79,18 @@ export function BuyTabContentProcess({ asset, onBack }: BuyTabContentProcessProp
...
+      .then((value) => toPaymentInformation(value))
       .then(setPaymentInfo);
   }, [validatedData]);
 
+  useEffect(() => {
+    if (selectedBankAccount && selectedBankAccount.preferredCurrency) {
+      setValue('currency', selectedBankAccount.preferredCurrency);
+    } else {
+      resetField('currency');
+    }
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply src/components/tabs/buy-tab-content/buy.process.tsx:91]: thought so, but we can remove it. no strong feelings

File: src/components/tabs/buy-tab-content/buy.process.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Krysh90) [reply src/components/tabs/buy-tab-content/buy.process.tsx:59]: and glitches the functionality, will revert

File: src/stories/StyledCoinListItem.tsx [hunk 5]
```diff
@@ -23,21 +24,30 @@ const NAME_MAPS: Record<AssetIconVariant, string> = {
...
+  disabled,
+  comingSoon,
+}: StyledCoinListItemProps) {
+  const name = comingSoon ? 'Coming soon' : NAME_MAPS[asset as AssetIconVariant];
   let buttonClasses = 'flex gap-2 rounded px-3 py-2 h-12';
 
-  disabled ? null : (buttonClasses += ' hover:bg-dfxGray-400/50 focus:bg-dfxGray-400/50 active:bg-dfxGray-400/80');
+  disabled || comingSoon
+    ? null
+    : (buttonClasses += ' hover:bg-dfxGray-400/50 focus:bg-dfxGray-400/50 active:bg-dfxGray-400/80');
```
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/stories/StyledCoinListItem.tsx:39]: Ok interesting

File: src/components/tabs/buy-tab-content/buy.process.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/components/tabs/buy-tab-content/buy.process.tsx:59]: Ok

**Review by davidleomay — APPROVED** [review]

**Review by niklasdfx — COMMENTED** [review]
*thumbs up*

**Pull request merged**